src/test/ui/async-await/no-std.rs

```diff
@@ -0,0 +1,13 @@
+// edition:2018
+// check-pass
+
+#![no_std]
+#![crate_type = "rlib"]
+
+use core::future::Future;
+
+async fn a(f: impl Future) {
+    f.await;
+}
+
+fn main() {}
```

Comment:
Could we test more like a whole program with more interactions?

Reply:
Please add more run-pass with:
* spawn onto the main thread with a basic waker vtable
* a spinlock holds that task

Reply:
There are already a lot of tests for async/await, most just don't turn on the `#![no_std]` flag. Since the underlying implementation is the same there is not really a difference though.

(I'm not entirely sure what the spinlock test would accomplish here though, you can store tasks however you want as long as you can get a `Pin<&mut Self>` to poll the future)

Reply:
> There are already a lot of tests for async/await, most just don't turn on the #![no_std] flag.

If we have a build matrix that enables no_std flag for async/await tests, it is ok to not add tests.(iirc we don't have) I assume compiler harness doesn't do that unless you explicitly ask for it. So when it is passing for std doesn't mean that it will pass by all cases of no_std.

> (I'm not entirely sure what the spinlock test would accomplish here though,

To demonstrate also it is working in no_std env without any extra dependency brought in.